ecs-cli/modules/cli/compose/entity/service/service.go

```diff
@@ -132,8 +145,23 @@ func (s *Service) LoadContext() error {
 		if loadBalancerName != "" {
 			s.loadBalancer.LoadBalancerName = aws.String(loadBalancerName)
 		}
+		s.loadBalancers = []*ecs.LoadBalancer{s.loadBalancer}
+		// s.loadBalancers = append(s.loadBalancers, s.loadBalancer)
 		s.role = role
 	}
+
+	if len(targetGroupsList) != 0 {
+		if role == "" {
+			return errors.Errorf("[--%s] is required when either when [--%s] flag is specified", flags.RoleFlag, flags.TargetGroupsFlag)
+		}
+		loadBalancers, err := utils.ParseLoadBalancers(targetGroupsList)
+		if err != nil {
+			return err
+		}
+		s.loadBalancers = append(s.loadBalancers, loadBalancers...)
```

Comment:
Will this line fail if only new flag is specified? Since `s.loadBalancers` is not initialized yet.

Reply:
Actually no, it is already initialized as an empty array

Reply:
yeah but that init might get skipped if the old flags are not specified

Reply:
So if only flags that are specified are the --role and --target-groups, it will not throw an error. Tbh I do not know the structure well enough to when it initializes it